pkg/corerp/renderers/aci/gateway/render.go

```diff
@@ -0,0 +1,397 @@
+		},
+		Properties: &armnetwork.PublicIPAddressPropertiesFormat{
+			PublicIPAddressVersion:   to.Ptr(armnetwork.IPVersionIPv4),
+			PublicIPAllocationMethod: to.Ptr(armnetwork.IPAllocationMethodStatic),
+			DNSSettings: &armnetwork.PublicIPAddressDNSSettings{
+				DomainNameLabel: to.Ptr(dnsPrefix),
+			},
+		},
+	}
+
+	publicIPResource := rpv1.OutputResource{
+		LocalID: rpv1.LocalIDAzurePublicIP,
+		ID:      resources.MustParse(resourceGroupID + "/providers/Microsoft.Network/publicIPAddresses/" + gateway.Name),
+		CreateResource: &rpv1.Resource{
+			ResourceType: resourcemodel.ResourceType{
+				Type:     "Microsoft.Network/publicIPAddresses",
+				Provider: resourcemodel.ProviderAzure,
+			},
+			Data: publicIP,
+		},
+	}
+
+	outputResources = append(outputResources, publicIPResource)
+
+	nsgName := gateway.Name + "-nsg"
+	nsg := &armnetwork.SecurityGroup{
+		Name:     to.Ptr(nsgName),
+		Location: to.Ptr(aciLocation),
+		Properties: &armnetwork.SecurityGroupPropertiesFormat{
+			SecurityRules: []*armnetwork.SecurityRule{
+				{
+					Name: to.Ptr("AppGatewayV2ProbeInbound"),
+					Properties: &armnetwork.SecurityRulePropertiesFormat{
+						Description:              to.Ptr("Allow traffic from GatewayManager. This rule is needed for application gateway probes to work."),
+						Protocol:                 to.Ptr(armnetwork.SecurityRuleProtocolTCP),
+						SourceAddressPrefix:      to.Ptr("GatewayManager"),
+						SourcePortRange:          to.Ptr("*"),
+						DestinationAddressPrefix: to.Ptr("*"),
+						DestinationPortRange:     to.Ptr("65200-65535"),
+						Access:                   to.Ptr(armnetwork.SecurityRuleAccessAllow),
+						Direction:                to.Ptr(armnetwork.SecurityRuleDirectionInbound),
+						Priority:                 to.Ptr[int32](100),
+					},
+				},
+				{
+					Name: to.Ptr("AllowHTTPInbound"),
+					Properties: &armnetwork.SecurityRulePropertiesFormat{
+						Description:              to.Ptr("Allow Internet traffic on port 80."),
+						Protocol:                 to.Ptr(armnetwork.SecurityRuleProtocolTCP),
+						SourceAddressPrefix:      to.Ptr("Internet"),
+						SourcePortRange:          to.Ptr("*"),
+						DestinationAddressPrefix: to.Ptr("*"),
+						DestinationPortRange:     to.Ptr(fmt.Sprintf("%d", targetPort)),
+						Access:                   to.Ptr(armnetwork.SecurityRuleAccessAllow),
+						Direction:                to.Ptr(armnetwork.SecurityRuleDirectionInbound),
+						Priority:                 to.Ptr[int32](110),
+					},
+				},
+				{
+					Name: to.Ptr("AllowPublicIPAddress"),
+					Properties: &armnetwork.SecurityRulePropertiesFormat{
+						Description:              to.Ptr("Allow traffic from public ip address."),
+						Protocol:                 to.Ptr(armnetwork.SecurityRuleProtocolTCP),
+						SourceAddressPrefix:      to.Ptr("Internet"),
+						SourcePortRange:          to.Ptr("*"),
+						DestinationAddressPrefix: to.Ptr("placeholder"),
```

Comment:
Is "placeholder" OK here?

Reply:
This value gets updated in the handler. I'll add a comment so it's clear